src/components/backend-ai-session-launcher.ts

```diff
@@ -2746,6 +2794,104 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
     this.shadowRoot.querySelector('#HPCOptimizationOptions').style.display = isOpenMPChecked ? 'none' : 'block';
   }
 
+  /**
+   * Toggle startup code input section according to session type
+   *
+   * @param {Event} e
+   */
+  _toggleStartUpCommandEditor(e) {
+    this.sessionType = e.target.value;
+    const isBatchmode: boolean = (this.sessionType === 'batch');
+    const startUpCommandEditor = this.shadowRoot.querySelector('#batch-mode-config-section');
+    startUpCommandEditor.style.display = isBatchmode ? 'inline-flex' : 'none';
+    if (isBatchmode) {
+      const editor = this.shadowRoot.querySelector('#command-editor');
+      editor.refresh();
+      editor.focus();
+    }
+  }
+
+  /**
+   * Toggle scheduling time when session type is in batch
+   *
+   */
+  _toggleScheduleTime() {
+    this.useScheduledTime = this.shadowRoot.querySelector('#use-scheduled-time').selected;
+    this.shadowRoot.querySelector('vaadin-date-time-picker').style.display = this.useScheduledTime ? 'block': 'none';
+    this._getSchedulableTime();
+  }
+
+  /**
+   * Returns schedulable time according to current time (default: 2min after current time)
+   *
+   * @return {string}
+   */
+  _getSchedulableTime() {
+    const getFormattedTime = (date) => {
+      // YYYY-MM-DD`T`hh:mm:ss
+      return date.getFullYear() + '-' + (date.getMonth()+1) + '-' + date.getDate() +
+            'T' + date.getHours() + ':' + date.getMinutes() + ':' + date.getSeconds();
+    };
+    const schedulerEl = this.shadowRoot.querySelector('vaadin-date-time-picker');
+    let currentTime = new Date();
+    const extraMinutes = 60 * 2 * 1000;
+    // add 2min
+    let futureTime = new Date(currentTime.getTime() + extraMinutes);
+    // disable scheduling in past
+    schedulerEl.min = getFormattedTime(currentTime);
+    // schedulerEl.value = getFormattedTime(futureTime);
+
+    if (schedulerEl.value && schedulerEl.value !== '') {
+      const scheduledTime = new Date(schedulerEl.value).getTime();
+      currentTime = new Date();
+      if (scheduledTime <= currentTime.getTime()) {
+        futureTime = new Date(currentTime.getTime() + extraMinutes);
+        schedulerEl.value = getFormattedTime(futureTime);
+      }
+    } else {
+      schedulerEl.value = getFormattedTime(futureTime);
+    }
+    this._setRelativeTimeStamp();
+
+    // interval every 1 sec.
+    setTimeout(() => {
+      this._getSchedulableTime();
+    }, 1000);
```

Comment:
This might be a potential memory leak and unresponsive UI.

`_getSchedulableTime` is called whenever the value of the date time picker AND the `#use-scheduled-time` switch change. This leads to the duplicated pending calls to this `setTimeout` which would not be the desired behavior. You may check this by inserting a debug message at the start of the `_getSchedulableTime` method and repeat turning on/off the `#use-scheduled-time`.

The `setTimeout` should be explicitly released whenever it is not used. Also, we need to ensure there is only one pending call to the `setTimeout` in this case. We don't need to call it when user did not enable the `#use-scheduled-time` switch.

Reply:
> I also think it is better to display the type of session in the session list grid even though this PR introduces the "Interactive" and "Batch" tabs. Users may want to distinguish the type of sessions in the "Running" and "Finished" tabs.

I agree. I'll add a column for session type.

Reply:
I replaced the recursive call of `_getSchedulableTime` with pausing/resuming `setInterval`. This will remarkably reduce calling `_getSchedulableTime` function.